### PR TITLE
MM-24864  Migrate API handler updateChannelPrivacy to be compatible with local mode

### PR DESCRIPTION
**Pull request by reflog**
#### Summary
Added local mode support for updateChannelPrivacy
#### Ticket Link
https://mattermost.atlassian.net/browse/MM-24864

### CODE REVIEW

File: api4/channel_test.go [hunk 1]
```diff
@@ -1485,39 +1485,40 @@ func TestUpdateChannelPrivacy(t *testing.T) {
...
 
-	for _, tc := range tt {
-		t.Run(tc.name, func(t *testing.T) {
-			_, resp := Client.UpdateChannelPrivacy(tc.channel.Id, tc.expectedPrivacy)
-			CheckBadRequestStatus(t, resp)
-		})
-	}
+		for _, tc := range tt {
+			t.Run(tc.name, func(t *testing.T) {
+				_, resp := Client.UpdateChannelPrivacy(tc.channel.Id, tc.expectedPrivacy)
```
Comment (mgdelacroix) [inline api4/channel_test.go:1498]: The client should be replaced with the variable injected by the helper:

```suggestion
				_, resp := client.UpdateChannelPrivacy(tc.channel.Id, tc.expectedPrivacy)
```

File: api4/channel_test.go [hunk 2]
```diff
@@ -1485,39 +1485,40 @@ func TestUpdateChannelPrivacy(t *testing.T) {
...
-			CheckNoError(t, resp)
-			assert.Equal(t, tc.expectedPrivacy, updatedChannel.Type)
-			updatedChannel, err := th.App.GetChannel(tc.channel.Id)
-			require.Nil(t, err)
-			assert.Equal(t, tc.expectedPrivacy, updatedChannel.Type)
-		})
-	}
+		for _, tc := range tt {
+			t.Run(tc.name, func(t *testing.T) {
+				updatedChannel, resp := Client.UpdateChannelPrivacy(tc.channel.Id, tc.expectedPrivacy)
```
Comment (mgdelacroix) [inline api4/channel_test.go:1513]: same here:
```suggestion
				updatedChannel, resp := client.UpdateChannelPrivacy(tc.channel.Id, tc.expectedPrivacy)
```

**Comment by reflog** [comment]
@mgdelacroix - Leaving this as DRAFT until we get a reply from PMs regarding this API

File: api4/channel_test.go [hunk 3]
```diff
@@ -1481,43 +1481,44 @@ func TestUpdateChannelPrivacy(t *testing.T) {
...
-			CheckNoError(t, resp)
-			assert.Equal(t, tc.expectedPrivacy, updatedChannel.Type)
-			updatedChannel, err := th.App.GetChannel(tc.channel.Id)
-			require.Nil(t, err)
-			assert.Equal(t, tc.expectedPrivacy, updatedChannel.Type)
-		})
-	}
+		for _, tc := range tt {
+			t.Run(tc.name, func(t *testing.T) {
+				updatedChannel, resp := c.UpdateChannelPrivacy(tc.channel.Id, tc.expectedPrivacy)
```
Comment (ashishbhate) [inline api4/channel_test.go:1513]: I think the `c` (`th.Client`) here needs to be changed to `client`
```suggestion
				updatedChannel, resp := client.UpdateChannelPrivacy(tc.channel.Id, tc.expectedPrivacy)
```

**Comment by reflog** [comment]
closing, this will need to be reworked

**Pull request closed without merging**